### PR TITLE
Upgrade dependencies to support future migration to Java 9.

### PR DESCRIPTION
**Pull request by millems**
1. Migrated from FindBugs to SpotBugs
2. Upgraded any dependencies that no longer worked on Java 9
3. Added explicit dependency on javax annotations for @Generated annotation access.
4. Fixed a warning from SpotBugs that was not caught by FindBugs.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: java
 jdk:
-  - oraclejdk8
+  - oraclejdk9
 sudo: true
 dist: precise
```
Comment (dagnir) [inline .travis.yml]: Think we also need to update to Trusty to get 9

**Comment by spfink** [comment]
What kind of testing have you done to verify this will work for all consumers?

**Comment by millems** [comment]
Talked about it, and it might be good to wait a while before upgrading to Java 9. I'd guess most testers are still on Java 8. Upgrading later on when more people have had a chance to upgrade is probably the better idea.

I'm going to tweak this PR to include everything BUT actually performing the java 9 upgrade. That'll make it a 1-character change to upgrade when we are ready.

**Comment by brharrington** [comment]
> Talked about it, and it might be good to wait a while before upgrading to Java 9.

Does this mean that the 2.x SDK will require jdk9 or higher at runtime? Since jdk9 is not considered a [long term stable release](http://www.oracle.com/technetwork/java/eol-135779.html), it seems like it would be good if it could run on jdk8 until at least 18.9 is out.

As a side note, given you are still supporting jdk8 for now, why not configure a travis matrix build that verifies changes on both jdk8 and jdk9? We have been [doing that](https://github.com/Netflix/spectator/blob/master/.travis.yml) and it has helped prevent a few regressions from coming in and breaking support for running on jdk9.

**Pull request merged**